stellar/processors.js

```diff
@@ -0,0 +1,215 @@
+                }),
+            ]);
+
+            return {
+                owner,
+                operator,
+                domainSeparator,
+                minimumRotationDelay,
+                previousSignersRetention,
+                initialSigners,
+            };
+        }
+
+        case 'InterchainTokenService': {
+            const gatewayAddress = nativeToScVal(Address.fromString(chain.contracts?.AxelarGateway?.address), { type: 'address' });
+            const gasServiceAddress = nativeToScVal(Address.fromString(chain.contracts?.AxelarGasService?.address), { type: 'address' });
+            const itsHubAddress = nativeToScVal(config.axelar?.contracts?.InterchainTokenService?.address, { type: 'string' });
+            const chainName = nativeToScVal(chain.axelarId, { type: 'string' });
+            const nativeTokenAddress = nativeToScVal(Address.fromString(chain?.tokenAddress), { type: 'address' });
+            const interchainTokenWasmHash = BytesToScVal(await uploadContract('InterchainToken', options, wallet, chain));
+            const tokenManagerWasmHash = BytesToScVal(await uploadContract('TokenManager', options, wallet, chain));
+
+            return {
+                owner,
+                operator,
+                gatewayAddress,
+                gasServiceAddress,
+                itsHubAddress,
+                chainName,
+                nativeTokenAddress,
+                interchainTokenWasmHash,
+                tokenManagerWasmHash,
+            };
+        }
+
+        case 'AxelarOperators':
+            return { owner };
+
+        case 'AxelarGasService': {
+            const operatorsAddress = chain.contracts?.AxelarOperators?.address;
+
+            validateParameters({
+                isValidStellarAddress: { operatorsAddress },
+            });
+
+            const operator = operatorsAddress ? nativeToScVal(Address.fromString(operatorsAddress), { type: 'address' }) : owner;
+
+            return { owner, operator };
+        }
+
+        case 'Upgrader': {
+            return {};
+        }
+
+        case 'AxelarExample': {
+            const gatewayAddress = nativeToScVal(Address.fromString(chain.contracts?.AxelarGateway?.address), { type: 'address' });
+            const gasServiceAddress = nativeToScVal(Address.fromString(chain.contracts?.AxelarGasService?.address), { type: 'address' });
+            const itsAddress = options.useDummyItsAddress
+                ? gatewayAddress
+                : nativeToScVal(chain.contracts?.InterchainTokenService?.address, { type: 'address' });
+
+            return { gatewayAddress, gasServiceAddress, itsAddress };
+        }
+
+        case 'Multicall': {
+            return {};
+        }
+
+        default:
+            throw new Error(`Unknown contract: ${contractName}`);
+    }
+};
+
+const uploadContract = async (contractName, options, wallet, chain) => {
+    const contractCodePath = await getContractCodePath(options, contractName);
```

Comment:
isn't the `preActionHook` doing this already?

Reply:
Getting the contract code path isn't specific to TM/IT. It's to allow fetching any contract. The TM/IT workaround from the helper can be deleted but the pre action hook still makes sense